### PR TITLE
OSASINFRA-3362: Fix CAPI bootstrap deletion

### PR DESCRIPTION
**Pull request by MaysaMacedo**
The tear down of local capi controller is happening
right after triggering the deletion of the
bootstrap machine, leaving almost no time
for the machine deletion event to be handled
resulting in a leftover bootstrap machine.
This commit fixes the issue by checking up to
2 minutes if the bootstrap machine was deleted
prior to destroy the local control plane.

### CODE REVIEW

File: pkg/infrastructure/clusterapi/clusterapi.go [hunk 1]
```diff
@@ -323,14 +324,39 @@ func (i *InfraProvider) DestroyBootstrap(dir string) error {
...
 			},
-		}); client.IgnoreNotFound(err) != nil {
+		}); err != nil {
 			return fmt.Errorf("failed to delete bootstrap machine: %w", err)
 		}
+
+		machineDeletionTimeout := 2 * time.Minute
+		logrus.Infof("Waiting up %v for machine deletion  %s/%s...", machineDeletionTimeout, machineNamespace, machineName)
+		machineContext, cancel := context.WithTimeout(context.TODO(), machineDeletionTimeout)
+		timer.StartTimer("Machine deletion")
```
Comment (patrickdillon) [inline pkg/infrastructure/clusterapi/clusterapi.go:341]: This timer seems like it is basically the same as the destroy bootstrap timer, as this is the only logic currently in destroy bootstrap. Maybe we don't need this?

**Review by patrickdillon — COMMENTED** [review]
Thanks for adding this!

File: pkg/infrastructure/clusterapi/clusterapi.go (hunk 1 above)
Comment (MaysaMacedo) [reply pkg/infrastructure/clusterapi/clusterapi.go]: Indeed

**Comment by jcpowermac** [comment]
LGTM from the perspective of vsphere

File: pkg/infrastructure/clusterapi/clusterapi.go [hunk 2]
```diff
@@ -323,14 +324,39 @@ func (i *InfraProvider) DestroyBootstrap(dir string) error {
...
+				Name:      machineName,
+				Namespace: machineNamespace,
 			},
-		}); client.IgnoreNotFound(err) != nil {
+		}); err != nil {
 			return fmt.Errorf("failed to delete bootstrap machine: %w", err)
 		}
+
+		machineDeletionTimeout := 2 * time.Minute
+		logrus.Infof("Waiting up %v for bootstrap machine deletion %s/%s...", machineDeletionTimeout, machineNamespace, machineName)
```
Comment (patrickdillon) [inline pkg/infrastructure/clusterapi/clusterapi.go:339]:
```suggestion
		logrus.Infof("Waiting up to %v for bootstrap machine deletion %s/%s...", machineDeletionTimeout, machineNamespace, machineName)
```

**Review by mandre — COMMENTED** [review]
/lgtm

This effectively deletes the bootstrap node, as shown in [the openstack logs](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_installer/8104/pull-ci-openshift-installer-master-altinfra-e2e-openstack-capi-ovn/1765524033744408576/artifacts/e2e-openstack-capi-ovn/openstack-gather/artifacts/openstack_nodes.log)

**Pull request merged**